_posts/2019-01-25-best-practices-for-building-web-appbuilder-widgets.md

```diff
@@ -9,18 +9,27 @@ categories:
+Recently, I have been working on a [project](https://github.com/agrc/wfrc-wab-widgets/) building custom Web AppBuilder (WAB) widgets for a client. This has been an interesting digression from our typical development projects, and working within the confines of WAB has proved to be both liberating and frustrating. In this post, I would like to share a few solutions that have helped me make the most of working with WAB.
+
+## The Pros and Cons of Working with WAB Widgets
+But before we get to those tips, let's talk about some of the general pros and cons of working with WAB that I have discovered through this project. First, being able to focus on a single widget and not having to worry about an entire application has been great. I have loved letting Esri take care of all of the basic application stuff, such as loading the application framework, web map, and widgets. However, not having control over the entire system has created some headaches. That's where the WAB solutions came in.
+
+## Tips for Building WAB Widgets
+
+### 1. Check out Esri's WAB-related Yeoman generators project.
 
 The first big win for me was finding [Esri's WAB-related Yeoman generators project](https://github.com/Esri/generator-esri-appbuilder-js). It provides several different generators. The first sets up a project structure and development environment specific to developing WAB widgets. It includes:
-* a best practice folder structure
-* grunt tasks for JS tooling including babel, livereload, and sass
-* a grunt task to automate the copying of your development code to WAB developer edition
+* a best practice folder structure;
+* grunt tasks for JS tooling, including babel, livereload, and sass; and
+* a grunt task to automate the copying of your development code to WAB developer edition.
 
 Esri also includes a generator for scaffolding out the widgets themselves. This project provided a great starting point for my work.
 
-Another great tip that I found was to write your WAB widgets as plain old dojo widgets (inherit from `dijit._WidgetBase`) and then wrap them for WAB. This made unit tests much easier. The other benefit to this approach is that it makes it much easier to reuse the widgets in non-WAB applications. Here's an example repo: [gbochenek/wab-test-example](https://github.com/gbochenek/wab-test-example).
+### 2. Write your WAB widgets as dojo widgets.
```

Comment:
Again, please just double check the subheadings and whether they're even accurate! ;) And if we don't like them, that's ok! I thought the subheadings were worth exploring though.